### ISSUE TITLE
Improve language in spatial overlap validation messages

### ISSUE THREAD

**Issue by davemfish** [natcap/invest#502]
@newtpatrol  suggests the current language isn't well-understood by users and we could do better:

>"Bounding box does not intersect at least one other [...]", and as we've seen on the forum, a lot of people don't understand what that means. Could it be changed to something like "Not all of the spatial layers overlap each other. Do they all have the same projected coordinate system?" 

I agree this proposed language is clearer.

**Comment by davemfish** [comment]
For this particular validation message, the workbench does some parsing & formatting to improve readability, so it will be important to make sure that still works well. There are workbench tests covering that formatting which will probably fail if the language is changed. So that will be a good reminder.

**Comment by dcdenu4** [comment]
I agree this is a tricky one. I wonder if something generic up front like Stacie has:
> Not all of the spatial layers overlap each other. Please check projections in a GIS tool as well as the requirements for the model.

And then also dump some helpful debugging info for developer troubleshooting? If we wanted to be fancy it could even be a `Details` dropdown that's hidden?
> Bounding box does not intersect at least one other [...]
